api/src/test/java/org/openmrs/notification/db/hibernate/HibernateAlertDAOTest.java

```diff
@@ -9,34 +9,98 @@
  */
 package org.openmrs.notification.db.hibernate;
 
+import org.junit.Assert;
 import org.junit.Before;
+
+import java.util.List;
+
+import org.hibernate.SessionFactory;
 import org.junit.Test;
+import org.openmrs.User;
 import org.openmrs.api.context.Context;
-import org.openmrs.notification.AlertService;
+import org.openmrs.notification.Alert;
 import org.openmrs.test.BaseContextSensitiveTest;
 
 /**
  * This class tests the hibernate alert data access. TODO Consider changing this and all subsequent
  * tests to use dbunit
  */
 public class HibernateAlertDAOTest extends BaseContextSensitiveTest {
+	private HibernateAlertDAO hibernateAlertDAO;
 	
 	@Before
 	public void runBeforeEachTest() {
+		Context.openSession();
 		authenticate();
+		hibernateAlertDAO = new HibernateAlertDAO();
+		hibernateAlertDAO.setSessionFactory((SessionFactory) applicationContext.getBean("sessionFactory"));
 	}
 	
+	/**
+	 * Creates and Returns alert with Id 1234
+	 * 
+	 * @return Alert object
+	 */
+
+	private Alert createAlert() {
+		Alert alert = new Alert(1234);
+		alert.setText("Testing the hibernate DAO");
+		return alert;
+
+	}
+
 	/**
 	 * Test that you can get alerts
 	 * 
 	 * @throws Exception
 	 */
 	@Test
-	public void shouldGetAlerts() {
-		
-		AlertService as = Context.getAlertService();
-		//System.out.println(as.getAllAlerts());
-		
+	public void getAlert_shouldReturnAlert() throws Exception {
+
+		Alert alert = createAlert();
+		Context.getAlertService().saveAlert(alert);
+		Alert savedAlert = Context.getAlertService().getAlert(1234);
+		Assert.assertEquals(savedAlert.getText(), "Testing the hibernate DAO");
+
+	}
+
+	@Test
+	public void saveAlert_shouldReturnNotNull() {
+
+		Alert alert = createAlert();
+		Context.getAlertService().saveAlert(alert);
+		Alert savedAlert = Context.getAlertService().getAlert(1234);
+		Assert.assertNotNull(savedAlert.getAlertId());
+
+	}
+
+	@Test
+
+	public void deleteAlert_shouldReturnNull() {
+
+		Alert alert = createAlert();
+		Context.getAlertService().saveAlert(alert);
+		hibernateAlertDAO.deleteAlert(alert);
+		Assert.assertNull(Context.getAlertService().getAlert(1234));
+
+	}
+
+	@Test
+
+	public void getAllAllerts_shouldReturnNotNull() {
+
+		List<Alert> allAllerts = hibernateAlertDAO.getAllAlerts(true);
+		Assert.assertNotNull(allAllerts);
+
+	}
+
+	@Test
+
+	public void getAlerts_shouldReturnNotNull() {
```

Comment:
@HerbertYiga Yo could make this more descriptive aswell as to what the method does such as `getAlerts_shouldReturnAlertsWhenUserIsPassedIn()` and better to assert on the number of alerts got back as opposed to it being null

Reply:
thanks alot @reagan-meant for this, what do other folks think around this cc @dkayiwa  @ibacher @mks-d @mseaton

Reply:
@HerbertYiga I'd agree with @reagan-meant here. Usually, I try to name unit tests something like:

```
<method>_should<ExpectedBehaviour>When<Conditions>
```

In other words, I should be able to tell what the test is supposed to test just by looking at the name of the test. That helps when the test fails, since it gives me a better clue why it might've failed or at least what behaviour I broke.